R/add_difference_row.R

```diff
@@ -0,0 +1,435 @@
+#' Add difference rows
+#'
+#' - [`add_difference_row.tbl_summary()`]
+#'
+#' @param x (`gtsummary`)\cr
+#'   Object with class 'gtsummary'
+#' @param ... Passed to other methods.
+#' @keywords internal
+#'
+#' @author Daniel D. Sjoberg
+#' @export
+add_difference_row <- function(x, ...) {
+  check_not_missing(x)
+  check_class(x, "gtsummary")
+  UseMethod("add_difference_row")
+}
+
+#' Add differences rows between groups
+#'
+#' @description
+#' `r lifecycle::badge('experimental')`\cr
+#' Adds difference to tables created by [`tbl_summary()`] as additional rows.
+#' This function is often useful when there are more than two groups to compare.
+#'
+#' Pairwise differences are calculated relative to the specified
+#' `by` variable's specified reference level.
+#'
+#' @inheritParams add_difference.tbl_summary
+#' @param reference (scalar)\cr
```

Comment:
maybe better `string`? Can it not be a string?

Reply:
Nice catch! I updated the documentation indicating to pass the character level for factors, and added a check that a scalar factor has not been passed.